common/src/test/java/org/opensearch/ml/common/TestHelper.java

```diff
@@ -26,15 +26,15 @@ public static <T> void testParse(ToXContentObject obj, Function<XContentParser,
     }
 
     public static <T> void testParse(ToXContentObject obj, Function<XContentParser, T> function, boolean wrapWithObject) throws IOException {
-        XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON);
+        XContentBuilder builder = XContentFactory.jsonBuilder();
```

Comment:
```suggestion
        XContentBuilder builder = MediaTypeRegistry.contentBuilder(XContentType.JSON);
```

Reply:
Just curious to know why we should prefer `MediaTypeRegistry.contentBuilder(XContentType.JSON);` over `XContentFactory.jsonBuilder();`? as both are doing the same.

```
    public static XContentBuilder jsonBuilder() throws IOException {
        return MediaTypeRegistry.contentBuilder(XContentType.JSON);
    }
```

Reply:
In this case it doesnt matter since we are using `XContentType.JSON` anyway. But incase we'd like to use a different `MediaType` the plugin is abstracted out about the format OpenSearch engine uses. So I would prefer `MediaType` where ever possible.

Reply:
With this change: https://github.com/opensearch-project/OpenSearch/pull/9156 it would make `XContentType.JSON` generic.

Reply:
I prefer to use the previous one because this is more like a syntactic sugar, and in ml-commons we support only json format for now, we can use `MediaTypeRegistry` in the future if we need to support different media type.

Reply:
> I prefer to use the previous one because this is more like a syntactic sugar,

I think this is fine for now. I was leaning toward eliminating `XContentFactory` altogether so as to not tempt a developer to use it in `opensearch-core` (and create the circular library dependency problem) but we already enforce no circular dependencies with the `opensearch-core` library. What we don't enforce are circular dependencies between other libraries. 

The intent is for `opensearch-core` to be the "center of gravity". For example, downstream plugin/extensions takes a required dependency on core and optional dependencies on the other libraries (e.g., x-content, geo, telementry, aggregations, mappers, replication, storage). This keeps the downstream opensearch dependencies lean. Why do we care? Because if someone wants to run an OpenSearch as a function (e.g., lambda, or cloud function) package size limits affect bootstrap times, class loading, and overall customer cost. Developers can afford to be more memory wreckless if the project only runs on self hosted, but load times and package sizes matter in cloud native deployments. This is why I'm so adamant on breaking down the monolith and not building a bloated comprehensive `opensearch-sdk`.